client/src/components/Admin/products/index.js

```diff
@@ -13,17 +13,57 @@ import DeleteIcon from "@mui/icons-material/Delete";
 import { DataGrid } from "@mui/x-data-grid";
 import Popup from "../common/popup";
 import ProductForm from "./productForm";
+import { useDispatch, useSelector } from "react-redux";
+import { deleteProduct, fetchProducts } from "../redux/productSlice";
 
 const Products = () => {
+  const dispatch = useDispatch();
   const [openPopup, setOpenPopup] = useState(false);
+  const [products, setProducts] = useState([]);
+  const [updateProduct, setUpdateProduct] = useState(null);
+
+  useEffect(() => {
+    dispatch(fetchProducts());
```

Comment:
add check for productsList.length is empty then only fetch products from the server